common/src/main/java/com/wynntils/features/user/CombatXPGainMessageFeature.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.notifications.NotificationManager;
+import com.wynntils.mc.event.ClientTickEvent;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.model.CharacterManager;
+import com.wynntils.wynn.utils.WynnUtils;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class CombatXPGainMessageFeature extends UserFeature {
+
+    @Config
+    public int tickDelay = 20;
+
+    private long tickCounter = 0;
+
+    private float newTickXP = 0;
+    private float lastTickXP = 0;
+    private float trackedPercentage = 0;
+    private int trackedLevel = 0;
+
+    @SubscribeEvent
+    public void onWorldStateChange(WorldStateEvent event) {
+        tickCounter = 0;
+        newTickXP = 0;
+        lastTickXP = 0;
+        trackedPercentage = 0;
+    }
+
+    @SubscribeEvent
+    public void onTick(ClientTickEvent.End event) {
+        if (!WynnUtils.onWorld()) return;
+
+        tickCounter++;
+
+        if (tickCounter % tickDelay != 0) return;
+
+        tickCounter = 0;
+
+        CharacterManager.CharacterInfo data = WynnUtils.getCharacterInfo();
```

Comment:
I think it could be a bit clearer if everything below this were extracted into a separate method, i.e. the things that actually are done when the ticks align.